### ISSUE TITLE
Tag Tracking Megathread - Grogu

### ISSUE THREAD

**Issue by dylangerdaly** [microg/GmsCore#1874]
**Is your feature request related to a problem? Please describe.**
Tag Tracking is coming to Android, this issue shall serve as a megathread for implementing Tag Tracking within microG.

Allowing the user to log _then_ report seen tags, or totally disable the entire feature would be a major plus.

Alternatively allowing the user to _just_ log seen tags would also be neat.

https://arstechnica.com/gadgets/2023/01/google-plans-airtag-clone-will-track-devices-with-3-billion-android-phones/
https://twitter.com/Za_Raczke/status/1615062461744549888

**Comment by alex9099** [comment]
Disable entirely may actually hurt one's privacy.
As "everyone" has a phone with play services, if a tag is with you, you'd more than likely be tracked without you knowing.

At least being alerted a tag has been with you for some time would be a "critical" feature

**Comment by alex9099** [comment]
Relevant project https://github.com/seemoo-lab/AirGuard/

**Comment by dylangerdaly** [comment]
Willing to start with a bounty for $200USD, as mentioned before this should fully implement Google's FindMy Network, allowing users to buy and use trackers with Google's FindMy Network, while giving microG users the ability to log requests sent to Google's backend, or disable it completely.

**Comment by freshgum-bubbles** [comment]
At the risk of being ants at a picnic, what would the legality of this look like?  Tapping into Google's proprietary network might result in a DMCA.  [I *especially* doubt Apple would be very happy](https://www.washingtonpost.com/technology/2020/07/24/apple-find-my-competition/) -- it would be wise to make sure this is actually legally feasible before going through the trouble of RE & building (and testing) it in microG.  Further, if something *is* done, perhaps doing it out-of-tree would be wise.

Nevertheless, if this *could* be implemented, I've love to see it.  Always wanted some sort of open-source Find My solution :-)

**Comment by RedAuburn** [comment]
https://www.techradar.com/phones/android/android-find-my-device

there looks to be a very big update to the FMD network incoming, including tracking items like Find My.

**Comment by LxtteDev** [comment]
Not being able to use Find My has been a major friction point for moving to any degoogled system.  Although Google having access to my location is a bit much, it's a negative I'm willing to live with.

**Comment by Idcrafter** [comment]
adding a Find My Device Feature to microG that works even if you're Phone is offline and also keeping privacy would be a great thing, it already has the location data and is mostly used and installed as system app. via ble would it be able to locate other devices if they are offline. 
it probably isn't a high priority to make such a system and hosting and managing such a backend for all that data is also not easy.

**Comment by rugk** [comment]
> hosting and managing such a backend for all that data is also not easy.

Wait no. As all other services microG uses, it would obviously (need to) use the official Google (or whatever the cross-platform thingy between Apple und Google is then/now called) backend. Otherwise, devices could not find each other…
After all, there is some similarity (and previous work/implementation here in microG) to the (covid) exposure notifications.

**Comment by Martin22** [comment]
> Has anyone had any luck reverse engineering the google findmy system?

Yes, there is a project: https://github.com/leonboe1/GoogleFindMyTools

**Comment by paolo-caroni** [comment]
> After all, there is some similarity (and previous work/implementation here in microG) to the (covid) exposure notifications.

@rugk please don't mix very different things. Covid exposue notifications was a beautiful and very privacy wise thing.
FindMy is a centralized spyng service, that promote to be encrypted even for google, so in theory google can't track you.

**Comment by CartCaved** [comment]
It seems one initial component of the Find Hub capability was merged into microg in April, in pr #3230, so there is some activity.

How can I join the bounty with another $50?
